tests/component-tests/pager/js/PagedTableTests.js

```diff
@@ -150,62 +159,99 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
-
-                if (link.hasClass(pager.pagerBar.options.styles.currentPage)) { // assumption that this will not change between now and then!
-                    assertVisibleTips("There shouldn't be any tooltips visible when the currentPage is focused", []);
-                } else {
-                    assertVisibleTips("Only the tooltip for page link " + (idx + 1) + ", in the " + location + " page bar is visible", [linkId]);
-                }
-                sequence.push({
-                    element: link,
-                    jQueryTrigger: "blur"
-                });
-                assertVisibleTips("There shouldn't be any tooltips visible when none of the pageLinks are focused", []);
-            };
-        };
+            sequence.push({
+                func: "fluid.tests.verifyLinksAndTooltips",
+                args: [pager, pageSize, linksList]
+            });
+        }
 
-        pageLinksTop.each(tooltipTest("top"));
-        pageLinksBottom.each(tooltipTest("bottom"));
         return {
             name: "Pager tooltip tests",
-            tests: {
-                name: "Tooltip visibility",
-                sequence: sequence
-            }
+            tests: [
+                {
+                    name: "Tooltip visibility and contents",
+                    sequence: sequence
+                }
+            ]
         };
     };
 
+    fluid.tests.getTooltipContents = function(listOfData, pageSize) {
+        var tooltipContents = [];
+        for (var i = 0; i < listOfData.length; i += pageSize) {
+            var tooltipContent = [];
+            var pageStartIndex = i;
+            var pageEndIndex = (i + pageSize -1) < listOfData.length ? (i + pageSize -1) : (listOfData.length - 1);
+            tooltipContent.push({nodeName: "b", nodeText: listOfData[pageStartIndex].text});
+            tooltipContent.push({nodeName: "b", nodeText: listOfData[pageEndIndex].text});
+            tooltipContents.push(tooltipContent);
+        }
+
+        return tooltipContents;
+    };
+
+    fluid.tests.verifyLinksAndTooltips = function(pager, pageSize, linksList) {
+        fluid.tests.setPageSize(pager, pageSize);
+        fluid.tests.verifyPagerBar(pager, pageSize, "top");
+        fluid.tests.verifyPagerBar(pager, pageSize, "bottom");
+        fluid.tests.verifyToolTipsOfPagerBar(pager, pageSize, "top", fluid.tests.getTooltipContents(linksList, pageSize));
+        fluid.tests.verifyToolTipsOfPagerBar(pager, 4, "bottom", fluid.tests.getTooltipContents(linksList, pageSize));
+    };
+
+    fluid.tests.assertVisibleTips = function (pager, message, targetIds, expectedTooltip) {
+        fluid.tests.tooltip.assertVisible(message, pager, targetIds, null, function (tooltip) {
+            jqUnit.assertNode(tooltip.length, expectedTooltip, $("b", tooltip));
+        });
+    };
+
+    fluid.tests.setPageSize = function (pager, pageSize) {
+        fluid.changeElementValue(pager.locate("pageSize"), pageSize);
+    };
```

Comment:
You don't need `fluid.tests.setPageSize` you can just use fluid.changeElementValue directly